query/security/ldap/src/test/groovy/org/codice/ddf/admin/ldap/discover/LdapQuerySpec.groovy

```diff
@@ -0,0 +1,242 @@
+import org.codice.ddf.admin.ldap.TestLdapServer
+import org.codice.ddf.admin.ldap.commons.LdapMessages
+import org.codice.ddf.admin.ldap.fields.LdapDistinguishedName
+import org.codice.ddf.admin.ldap.fields.connection.LdapBindMethod
+import org.codice.ddf.admin.ldap.fields.connection.LdapBindUserInfo
+import org.codice.ddf.admin.ldap.fields.connection.LdapConnectionField
+import org.codice.ddf.admin.ldap.fields.connection.LdapEncryptionMethodField
+import org.codice.ddf.admin.ldap.fields.query.LdapQueryField
+import spock.lang.Specification
+
+import static org.codice.ddf.admin.ldap.LdapTestingCommons.noEncryptionLdapConnectionInfo
+import static org.codice.ddf.admin.ldap.LdapTestingCommons.simpleBindInfo
+
+class LdapQuerySpec extends Specification {
+    static TestLdapServer server
+    Map<String, Object> args
+    LdapQuery ldapQueryFunction
+
+    def setupSpec() {
+        server = TestLdapServer.getInstance().useSimpleAuth()
+        server.startListening()
+    }
+
+    def cleanupSpec() {
+        server.shutdown()
+        server = null
+    }
+
+    def setup() {
+        LdapTestingCommons.loadLdapTestProperties()
+        ldapQueryFunction = new LdapQuery()
+    }
+
+    def 'Fail on missing required fields'() {
+        setup:
+        def baseMsg = [LdapQuery.FIELD_NAME, BaseFunctionField.ARGUMENT]
+        def missingHostMsgPath = baseMsg + [LdapConnectionField.DEFAULT_FIELD_NAME, HostnameField.DEFAULT_FIELD_NAME]
+        def missingPortMsgPath = baseMsg + [LdapConnectionField.DEFAULT_FIELD_NAME, PortField.DEFAULT_FIELD_NAME]
+        def missingEncryptMsgPath = baseMsg + [LdapConnectionField.DEFAULT_FIELD_NAME, LdapEncryptionMethodField.DEFAULT_FIELD_NAME]
+        def missingUsernameMsgPath = baseMsg + [LdapBindUserInfo.DEFAULT_FIELD_NAME, CredentialsField.DEFAULT_FIELD_NAME, CredentialsField.USERNAME_FIELD_NAME]
+        def missingUserpasswordMsgPath = baseMsg + [LdapBindUserInfo.DEFAULT_FIELD_NAME, CredentialsField.DEFAULT_FIELD_NAME, CredentialsField.PASSWORD_FIELD_NAME]
+        def missingBindMethodMsgPath = baseMsg + [LdapBindUserInfo.DEFAULT_FIELD_NAME, LdapBindMethod.DEFAULT_FIELD_NAME]
+        def missingQueryBaseMsgPath = baseMsg + [LdapQuery.QUERY_BASE_FIELD_NAME]
+        def missingQueryMsgPath = baseMsg + [LdapQueryField.DEFAULT_FIELD_NAME]
+
+        when:
+        FunctionReport report = ldapQueryFunction.getValue()
+
+        then:
+        report.messages().size() == 8
+        report.messages().count {
+            it.getCode() == DefaultMessages.MISSING_REQUIRED_FIELD
+        } == 8
+
+        report.messages()*.getPath() as Set == [missingHostMsgPath, missingPortMsgPath, missingEncryptMsgPath,
+                                                missingUsernameMsgPath, missingUserpasswordMsgPath, missingBindMethodMsgPath,
+                                                missingQueryBaseMsgPath, missingQueryMsgPath] as Set
+    }
+
+    def 'Fail to connect to LDAP'() {
```

Comment:
I believe it is relevant to querying, it's ensuring the error messages are propagating correctly.